packages/babel-helper-define-polyfill-provider/src/types.ts

```diff
@@ -0,0 +1,108 @@
+import type { NodePath } from "@babel/traverse";
+import { types as t } from "@babel/core";
+import createMetaResolver from "./meta-resolver";
+
+type ObjectMap<T> = { [k: string]: T };
+
+export type Pattern = string | RegExp;
+
+export type MissingDependenciesOption =
+  | false
+  | {
+      log?: "per-file" | "deferred";
+      inject?: "import" | "throw";
+      // When true, log all the polyfills without checking if they are installed
+      all?: boolean;
+    };
+
+export type ProviderOptions<Opts = {}> = {
+  include?: Pattern[];
+  exclude?: Pattern[];
+} & Opts;
+
+export type PluginOptions = {
+  method: MethodString;
+  targets?: {
+    browsers: string | string[];
+    [target: string]: string | string[] | number;
+  };
```

Comment:
modified this [as mentioned](https://github.com/babel/babel-polyfills/pull/121#discussion_r838015118), but per that discussion, not sure if this is the intended type

Reply:
TS complains but then it seems to correctly pick up the types:

![TS playground screenshot](https://user-images.githubusercontent.com/7000710/161449923-b7af31dc-d5d4-426a-8a95-d1fbd27adf22.png)

I'll just put a `@ts-expect-error` to suppress it.

Reply:
Weird 👀  👀 , I'm surprised that works. I wonder if the unsound-ness might affect some other type resolution 🤔